### PR TITLE
make dnceng tools non packable

### PR DESCRIPTION
**Pull request by dkurepa**
issue: https://github.com/dotnet/arcade/issues/13650
Validation PR run: https://dev.azure.com/dnceng/internal/_build/results?buildId=2186664&view=results

### CODE REVIEW

**Comment by dkurepa** [comment]
Monitoring.Sdk is still getting published. I might be because of  `<PackageType>MSBuildSdk</PackageType>` will try to remove

**Comment by dkurepa** [comment]
I don't think we will. Dotnet.Status.Web doesn't use those packages, and none of the other projects here are used for deployments. To be safe, I can disable the rollouts for all internal tools in this PR too, we planned on doing that anyway

File: eng/deploy.yaml [hunk 1]
```diff
@@ -56,33 +56,14 @@ stages:
...
-  dependsOn:
-  - build
-  - Validate
-  - approval
-  jobs:
-  - job: notifyStartDeployment
-    displayName: Notify deployment start
-    pool: server
-    steps:
-    - task: InvokeRESTAPI@1
```
Comment (dkurepa) [inline eng/deploy.yaml:71]: I'm not sure if this is OK to remove, since it might be used for Maestro deployments too?

Comment (missymessa) [reply eng/deploy.yaml:71]: I think this was at least necessary for RolloutScorer. Let me know if ProdCon isn't planning to use results from RolloutScorer anymore for Arcade Services and I'll make sure it gets removed from RolloutScorer.

**Review by riarenas — APPROVED** [review]
Looks reasonable to me!

**Pull request merged**